project-base/tests/ShopBundle/Functional/Model/Product/Search/FilterQueryTest.php

```diff
@@ -0,0 +1,172 @@
+        $this->assertIdWithFilter($priceAscFilter, [40, 33, 50, 39, 29, 25, 26, 27, 28, 72], 'price asc');
+
+        $priceDescFilter = $filter->applyOrdering(ProductListOrderingConfig::ORDER_BY_PRICE_DESC, $pricingGroup);
+        $this->assertIdWithFilter($priceDescFilter, [72, 28, 27, 26, 25, 29, 39, 50, 33, 40], 'price desc');
+    }
+
+    public function testMatchQuery(): void
+    {
+        $filter = $this->createFilter();
+
+        $kittyFilter = $filter->search('kitty');
+        $this->assertIdWithFilter($kittyFilter, [1, 102, 101]);
+
+        $mg3550Filer = $filter->search('mg3550');
+        $this->assertIdWithFilter($mg3550Filer, [9, 144, 10, 145]);
+    }
+
+    public function testPagination(): void
+    {
+        $filter = $this->createFilter()
+            ->filterByCategory([9])
+            ->applyDefaultOrdering();
+
+        $this->assertIdWithFilter($filter, [72, 25, 27, 29, 28, 26, 50, 33, 39, 40]);
+
+        $limit5Filter = $filter->setLimit(5);
+        $this->assertIdWithFilter($limit5Filter, [72, 25, 27, 29, 28]);
+
+        $limit1Filter = $filter->setLimit(1);
+        $this->assertIdWithFilter($limit1Filter, [72]);
+
+        $limit4Page2Filter = $filter->setLimit(4)
+            ->setPage(2);
+        $this->assertIdWithFilter($limit4Page2Filter, [28, 26, 50, 33]);
+
+        $limit4Page3Filter = $filter->setLimit(4)
+            ->setPage(3);
+        $this->assertIdWithFilter($limit4Page3Filter, [39, 40]);
+
+        $limit4Page4Filter = $filter->setLimit(4)
+            ->setPage(4);
+        $this->assertIdWithFilter($limit4Page4Filter, []);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery $filterQuery
+     * @param int[] $ids
+     * @param string $message
+     */
+    protected function assertIdWithFilter(FilterQuery $filterQuery, array $ids, string $message = ''): void
+    {
+        /** @var \Elasticsearch\Client $es */
+        $es = $this->getContainer()->get(Client::class);
+
+        $params = $filterQuery->getQuery();
+
+        $params['_source'] = false;
+
+        $result = $es->search($params);
+        $this->assertSame($ids, $this->extractIds($result), $message);
+    }
+
+    /**
+     * @param array $result
+     * @return int[]
+     */
+    protected function extractIds(array $result): array
+    {
+        $hits = $result['hits']['hits'];
+
+        return array_map(static function ($element) {
+            return (int)$element['_id'];
+        }, $hits);
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery
+     */
+    protected function createFilter(): FilterQuery
+    {
+        /** @var \Shopsys\FrameworkBundle\Model\Product\Search\FilterQueryFactory $filterQueryFactory */
+        $filterQueryFactory = $this->getContainer()->get(FilterQueryFactory::class);
+        $filter = $filterQueryFactory->create(self::ELASTICSEARCH_INDEX);
```

Comment:
Hello, that can be a problem, if you are using Elasticsearch prefix in the project. Our tests failed because of this. Also, it can be incompatible in 8.0, where prefixes will be creating in tests environment. See (https://github.com/shopsys/shopsys/pull/933).

@grossmannmartin, @PetrHeinz

Reply:
Fixed in #1082, released as part of `v7.2.1`, thanks for reporting!